src/files-and-videos/videos-page/data/thunks.js

```diff
@@ -103,42 +101,47 @@ export function deleteVideoFile(courseId, id) {
-      const errors = await uploadVideo(
-        courseId,
-        uploadUrl,
-        file,
-        edxVideoId,
-      );
-      const { videos } = await fetchVideoList(courseId);
-      const newVideos = videos.filter(video => !videoIds.includes(video.edxVideoId));
-      const parsedVideos = updateFileValues(newVideos, true);
-      dispatch(addModels({
-        modelType: 'videos',
-        models: parsedVideos,
-      }));
-      dispatch(addVideoSuccess({
-        videoId: edxVideoId,
-      }));
-      dispatch(updateEditStatus({ editType: 'add', status: RequestStatus.SUCCESSFUL }));
-      if (!isEmpty(errors)) {
-        errors.forEach(error => {
-          dispatch(updateErrors({ error: 'add', message: error }));
-        });
+      const createUrlResponse = await addVideo(courseId, file);
+      // eslint-disable-next-line
+      console.log(`Post Response: ${createUrlResponse}`);
+      if (createUrlResponse.status < 200 && createUrlResponse.status >= 300) {
+        dispatch(updateErrors({ error: 'add', message: `Failed to add ${file.name}.` }));
+        dispatch(updateEditStatus({ editType: 'add', status: RequestStatus.FAILED }));
+        return;
+      }
+      const { edxVideoId, uploadUrl } = camelCaseObject(createUrlResponse.data).files[0];
+      const putToServerResponse = await uploadVideo(uploadUrl, file);
+      if (putToServerResponse.status < 200 || putToServerResponse.status >= 300) {
+        dispatch(updateErrors({ error: 'add', message: `Failed to upload ${file.name}.` }));
+        sendVideoUploadStatus(courseId, edxVideoId, 'Upload failed', 'upload_failed');
         dispatch(updateEditStatus({ editType: 'add', status: RequestStatus.FAILED }));
+        return;
       }
+      sendVideoUploadStatus(courseId, edxVideoId, 'Upload completed', 'upload_completed');
     } catch (error) {
       if (error.response && error.response.status === 413) {
         const message = error.response.data.error;
         dispatch(updateErrors({ error: 'add', message }));
       } else {
-        dispatch(updateErrors({ error: 'add', message: `Failed to add ${file.name}.` }));
+        dispatch(updateErrors({ error: 'add', message: `Failed to upload ${file.name}.` }));
       }
       dispatch(updateEditStatus({ editType: 'add', status: RequestStatus.FAILED }));
+      return;
+    }
+    try {
+      const { videos } = await fetchVideoList(courseId);
+      const newVideos = videos.filter(video => !videoIds.includes(video.edxVideoId));
+      const newVideoIds = newVideos.map(video => video.edxVideoId);
+      const parsedVideos = updateFileValues(newVideos, true);
+      dispatch(addModels({ modelType: 'videos', models: parsedVideos }));
+      dispatch(setVideoIds({ videoIds: videoIds.concat(newVideoIds) }));
+    } catch (error) {
+      dispatch(updateEditStatus({ editType: 'add', status: RequestStatus.FAILED }));
+      dispatch(updateErrors({ error: 'add', message: error.message }));
```

Comment:
Will error always have a message? If yes, is the message the error that we want to show the user?

Reply:
Going to show the FetchVideos error message to users, and log the actual error message.